dvc/utils/fs.py

```diff
@@ -20,15 +20,21 @@ def get_inode(path):
     return inode
 
 
-def get_mtime_and_size(path):
-    stat = os.stat(path)
-    size = stat.st_size
-    mtime = stat.st_mtime
+def get_mtime_signature_and_size(path, ignore_file_handler=None):
+    base_stat = os.stat(path)
+    size = base_stat.st_size
 
     if os.path.isdir(path):
-        for root, dirs, files in dvc_walk(path):
-            for name in dirs + files:
-                entry = os.path.join(root, name)
+        files_mtimes = {}
+        for root, dirs, files in dvc_walk(
+            str(path), ignore_file_handler=ignore_file_handler
```

Comment:
`str(path)` is an error. The rule is never pass path stringification to file utils, we should use `fspath()` or make `dvc_walk()` accept Path-like objects.

If that is done to force unicode then why it wasn't there already? As far as I remember @efiop removed that stringification.